### PR TITLE
Y24 154 pacbio pool volume check

### PR DESCRIPTION
**Pull request by seenanair**
Closes (https://github.com/sanger/traction-ui/issues/1784)

#### Changes proposed in this pull request
- Initial volume changes
- Adds available volume badge for pool volume on pool edit page

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

**Review by BenTopping — CHANGES_REQUESTED** [review]
Looks fine, haven't tested locally.
Could you add a test to check the volume badge exists (either in cypress or unit) and that you get the appropriate error/behaviour when you try and set the volume to an invalid amount. For example a test where it doesn't allow the pool to be updated unless the volume is valid.

**Comment by seenanair** [comment]
> Looks fine, haven't tested locally. Could you add a test to check the volume badge exists (either in cypress or unit) and that you get the appropriate error/behaviour when you try and set the volume to an invalid amount. For example a test where it doesn't allow the pool to be updated unless the volume is valid.

Yes, the first one is already there in unit test for PacbioPoolEdit which checks for a volume badge.

File: src/components/pacbio/PacbioPoolEdit.vue [hunk 1]
```diff
@@ -127,6 +151,9 @@ import { usePacbioPoolCreateStore } from '@/stores/pacbioPoolCreate.js'
 import useAlert from '@/composables/useAlert.js'
 import { ref, computed } from 'vue'
 import { eachRecord } from '@/lib/csv/pacbio.js'
+import TractionInfoIcon from '@/components/shared/icons/TractionInfoIcon.vue'
+import TractionTooltip from '@/components/shared/TractionTooltip.vue'
+import TractionBadge from '@/components/shared/TractionBadge.vue'
```
[the remark this answers is not in the document]
Comment (seenanair) [reply src/components/pacbio/PacbioPoolEdit.vue:156]: > Should these imports be include in src/components/shared/index.js so they can globally registered and not imported into each file?

Good point 👍

**Comment by seenanair** [comment]
@BenTopping I have added inline validation for pool fields as well to match with how the aliquot fields behave on editing(/typing) on the same page

File: tests/unit/components/pacbio/PacbioPoolEdit.spec.js [hunk 2]
```diff
@@ -245,6 +280,52 @@ describe('pacbioPoolEdit#edit', () => {
...
+    })
+    it('displays an error when insert size is not present', async () => {
+      const input = wrapper.find('[data-attribute=insert-size]')
+      await input.setValue('')
+      await nextTick()
+      expect(wrapper.find('[data-attribute=pool-insert_size-error]').text()).toContain(
+        'must be present',
+      )
+    })
+    it('displays an error when insert size is not present', async () => {
```
Comment (BenTopping) [inline tests/unit/components/pacbio/PacbioPoolEdit.spec.js:315]: 'displays an error when volume is not present', same name change required for concentration test above

**Review by BenTopping — APPROVED** [review]

File: tests/unit/components/pacbio/PacbioPoolEdit.spec.js (hunk 2 above)
Comment (seenanair) [reply tests/unit/components/pacbio/PacbioPoolEdit.spec.js]: Fixed. Copy paste error 🤦‍♀️

**Review by BenTopping — APPROVED** [review]

**Pull request merged**